packages/shared/src/common/date.ts

```diff
@@ -108,3 +109,66 @@ export function sortDate(
     ? buildDayjs(date1).diff(buildDayjs(date2))
     : buildDayjs(date2).diff(buildDayjs(date1));
 }
+
+export function throwIfInvalidHl7Timestamp(input: string): boolean {
+  if (!input || input.length !== 14) {
+    throw new MetriportError("Invalid HL7 date string format: expected YYYYMMDDHHMMSS");
+  }
+
+  if (!/^\d{14}$/.test(input)) {
+    throw new MetriportError("Invalid HL7 date string: must contain only digits");
```

Comment:
ditto BadRequestError, here and below in the same funcion

Reply:
addressed